src/main/java/com/zendesk/maxwell/RowMap.java

```diff
@@ -107,28 +136,12 @@ public String toJSON() throws IOException {
 		if ( this.txCommit )
 			g.writeBooleanField("commit", true);
 
-		g.writeObjectFieldStart("data"); // start of data: {
-
-		/* TODO: maintain ordering of fields in column order */
-		for ( String key: this.data.keySet() ) {
-			Object data = this.data.get(key);
-
-			if ( data == null )
-				continue;
-
-			if ( data instanceof List) { // sets come back from .asJSON as lists, and jackson can't deal with lists natively.
-				List<String> stringList = (List<String>) data;
+		writeMapToJSON("data", this.data, false);
 
-				g.writeArrayFieldStart(key);
-				for ( String s : stringList )  {
-					g.writeString(s);
-				}
-				g.writeEndArray();
-			} else {
-				g.writeObjectField(key, data);
-			}
+		if ( !this.oldData.isEmpty()) {
+			writeMapToJSON("old", this.oldData, true);
```

Comment:
hmm.  Ok, yeah, I guess I buy that if the value changed from NULL -> something we should explicitly say `old: { old_field: null }`.